internal/services/netapp/netapp_volume_helper.go

```diff
@@ -275,6 +276,58 @@ func expandNetAppVolumeDataProtectionSnapshotPolicyPatch(input []interface{}) *v
 	}
 }
 
+func expandNetAppVolumeDataProtectionBackupPolicy(input []interface{}) *volumes.VolumePropertiesDataProtection {
+	if len(input) == 0 {
+		return &volumes.VolumePropertiesDataProtection{}
+	}
+
+	backupPolicyObject := volumes.VolumeBackupProperties{}
+
+	backupRaw := input[0].(map[string]interface{})
+
+	if v, ok := backupRaw["backup_policy_id"]; ok {
+		backupPolicyObject.BackupPolicyId = utils.String(v.(string))
+	}
+
+	if v, ok := backupRaw["policy_enforced"]; ok {
+		backupPolicyObject.PolicyEnforced = utils.Bool(v.(bool))
+	}
+
+	if v, ok := backupRaw["backup_vault_id"]; ok {
+		backupPolicyObject.BackupVaultId = utils.String(v.(string))
+	}
+
+	return &volumes.VolumePropertiesDataProtection{
+		Backup: &backupPolicyObject,
+	}
+}
+
+func expandNetAppVolumeDataProtectionBackupPolicyPatch(input []interface{}) *volumes.VolumePatchPropertiesDataProtection {
+	if len(input) == 0 {
+		return &volumes.VolumePatchPropertiesDataProtection{}
+	}
+
+	backupPolicyObject := volumes.VolumeBackupProperties{}
+
+	backupRaw := input[0].(map[string]interface{})
```

Comment:
This might be nil so we should check to make sure it isn't when we check the length